src/platform/nxp/common/OTAImageProcessorImpl.cpp

```diff
@@ -279,8 +281,10 @@ void OTAImageProcessorImpl::HandleApply(intptr_t context)
      * This should be done with a delay so the device has enough time to send
      * the state-transition event when applying the update.
      */
-    ChipLogProgress(SoftwareUpdate, "Restarting device in 5 seconds ...");
-    DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds32(5 * 1000), HandleRestart, nullptr);
+    ChipLogProgress(SoftwareUpdate, "Restarting device, will reboot in %d seconds ...", imageProcessor->mDelayBeforeRebootSec);
```

Comment:
Technically we add deltaRebootDelayMs to the delay ... do we just rely it is very small? If so small that insignificant, why do we have it?

Reply:
When I did this modification, I thought having a small delta is better to make sure the report is sent, I tested with it and it worked fine. I prefer keeping it.